### PR TITLE
Fix monasca installer

### PR DESCRIPTION
**Pull request by jgrassler**
This pull request consists of two commits:

1) Various improvements to how monasca-installer is run.
2) A Cloud 8 specific fix for https://bugzilla.suse.com/show_bug.cgi?id=1096043

(1) can and should be backported to Cloud 7.

### CODE REVIEW

**Comment by jgrassler** [comment]
Note: please do not merge this, yet. I'd like to put it through its paces on a fresh cloud first. I'll update on here once that's done.

**Comment by jgrassler** [comment]
There was a missing `action :nothing` which is fixed now. I'll give it another spin on a fresh cloud...

**Comment by jgrassler** [comment]
Ok. This took some more doing. A considerable amount with lots of false starts, actually. I think it's working now, but I'll call it a weekend now and give it another go on a fresh cloud on Monday (I'll also clean up the history then).

**Comment by jgrassler** [comment]
Ok, I found a few more corner cases but these should be fixed now. As far as I'm concerned this is ready to merge.

File: chef/cookbooks/monasca/templates/default/run-monasca-installer.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+cd "/opt/monasca-installer"
+rm -f <%= @lock_file %>
+  /usr/bin/ansible-playbook \
+    -i monasca-hosts \
+    -e '@/opt/monasca-installer/crowbar_vars.yml' \
+    monasca.yml -vvv
+
+# Record version information to indicate a successful run.
+rpm -qa | grep -e crowbar-openstack -e monasca-installer | sort > <%= @lock_file %>
```
Comment (dirkmueller) [inline chef/cookbooks/monasca/templates/default/run-monasca-installer.erb]: useless use of grep. better do something like

LC_ALL=C rpm -qa crowbar-openstack \* monasca-installer | sort > ..

Comment (jgrassler) [reply chef/cookbooks/monasca/templates/default/run-monasca-installer.erb]: Fixed.

File: chef/cookbooks/monasca/recipes/master.rb [hunk 2]
```diff
@@ -155,17 +146,28 @@
   source "monasca-installer.logrotate"
 end
 
-ansible_cmd =
-  "rm -f #{lock_file} " \
-  "&& ansible-playbook " \
-    "-i monasca-hosts -e '@/opt/monasca-installer/crowbar_vars.yml' " \
-    "monasca.yml -vvv >> /var/log/monasca-installer.log 2>&1 " \
-  "&& echo '#{actual_versions}' > #{lock_file}"
+template "/usr/sbin/run-monasca-installer" do
```
Comment (dirkmueller) [inline chef/cookbooks/monasca/recipes/master.rb]: personally I'd prefer this to be in /usr/local/ ... to ensure there is no potential conflict with packaged stuff.

File: chef/cookbooks/monasca/recipes/master.rb [hunk 3]
```diff
@@ -155,17 +146,28 @@
...
+execute "remove lock file" do
+  command "rm -f #{lock_file}"
+  action :nothing
+end
 
 execute "run ansible" do
-  command ansible_cmd
-  cwd "/opt/monasca-installer"
-  action :nothing unless actual_versions != previous_versions
+  command "/usr/sbin/run-monasca-installer 2>&1"\
```
Comment (dirkmueller) [inline chef/cookbooks/monasca/recipes/master.rb]: in case you adjust the location, this needs corresponding adjustment

**Pull request merged**